app/main/views/suppliers.py

```diff
@@ -88,6 +90,114 @@ def list_suppliers():
+        lazyload(SupplierFramework.framework_agreements),
+    ).order_by(
+        Supplier.supplier_id
+    ).all()
+
+    supplier_rows = []
+
+    suppliers_with_a_complete_service = frozenset(framework.get_supplier_ids_for_completed_service())
+
+    for sf, supplier, ci in suppliers_and_framework:
+        declaration_status = sf.declaration.get('status') if sf.declaration else 'unstarted'
+        # For G10 we need to check the suppliers company detatils have been confirmed
+        company_details_confirmed = supplier.company_details_confirmed
+        application_status = 'application' if (
+            declaration_status == 'complete' and
+            supplier.supplier_id in suppliers_with_a_complete_service and
+            company_details_confirmed
+        ) else 'no_application'
+        application_result = ''
+        framework_agreement = False
+        variations_agreed = ''
+
+        if framework.status != 'open':
+            if sf.on_framework is None:
+                application_result = 'no result'
+            else:
+                application_result = 'pass' if sf.on_framework else 'fail'
+            framework_agreement = bool(getattr(sf.current_framework_agreement, 'signed_agreement_returned_at', None))
+            variations_agreed = ', '.join(sf.agreed_variations.keys()) if sf.agreed_variations else ''
+
+        supplier_rows.append({
+            "supplier_id": supplier.supplier_id,
+            "supplier_name": supplier.name,
+            "supplier_organisation_size": supplier.organisation_size,
+            "duns_number": supplier.duns_number,
+            "registered_name": supplier.registered_name,
+            "companies_house_number": supplier.companies_house_number,
+            # TODO: framework application result, framework agreement, variations agreed
+            'application_result': application_result,
+            'application_status': application_status,
+            'declaration_status': declaration_status,
+            'framework_agreement': framework_agreement,
+            'variations_agreed': variations_agreed,
+            # TODO: service counts for each lot
+            "published_services_count": {
+                lot_slugs_by_id[lot_id]: service_counts_by_lot_by_supplier.get(
+                    supplier.supplier_id, {}
+                ).get(lot_id, 0)
+                for lot_id in lot_slugs_by_id.keys()
+            },
+            "contact_information": {
+                'contact_name': ci.contact_name,
+                'contact_email': ci.email,
+                'contact_phone_number': ci.phone_number,
+                'address_first_line': ci.address1,
+                'address_city': ci.city,
+                'address_postcode': ci.postcode,
+                'address_country': supplier.registration_country,
+            }
+        })
+
+    return jsonify(suppliers=[supplier for supplier in supplier_rows]), 200
```

Comment:
What does this list comprehension do? Can't you just use `supplier_rows` directly?

Reply:
That is an extremely good question. We copied the pattern from the equivalent `users` export, which has been like that for... 3 years? https://github.com/alphagov/digitalmarketplace-api/commit/5dc89f003fe6a90346119487d6b5d02499c04933#diff-1b3a761e88acb979c136b48cf15b3976

Reply:
I've removed them in both export views.